Source/WebCore/Modules/gamepad/NavigatorGamepad.cpp

```diff
@@ -91,6 +91,9 @@ const Vector<RefPtr<Gamepad>>& NavigatorGamepad::getGamepads(Navigator& navigato
 
     }
 
+    if (!isFeaturePolicyAllowedByDocumentAndAllOwners(FeaturePolicy::Type::Gamepad, *document, LogFeaturePolicyFailure::Yes))
+        return Exception { ExceptionCode::SecurityError, "Third-party iframes are not allowed to call getGamepads() unless explicitly allowed via Feature-Policy (gamepad)"_s };
```

Comment:
This follows the spec so is ok as is.
The specs using permission policy are not throwing/rejecting with the same error (sometimes SecurityError, sometimes NotAllowedError for instance).

Reply:
Yeah, I did file an issue about this a few years back... unfortunately, it hasn't been addressed yet: https://github.com/w3c/webappsec-permissions-policy/issues/396